### PR TITLE
[CSL-1305] Allow to use different genesis.bin depending on DCONFIG

### PR DESCRIPTION
**Pull request by georgeee**
* Allow to specify genesis.bin as constant
* Get rid off aggregates: prod, wallet

### CODE REVIEW

File: core/constants.yaml [hunk 1]
```diff
@@ -121,31 +123,38 @@ qanet: &qanet
 
   recoveryHeadersMessage: 100
 
+  genesisBinSuffix: qa
+
 testnet: &testnet
   <<: *prod_base
 
-  k: 60
+  k: 72
```
Comment (georgeee) [inline core/constants.yaml]: This change is not related to this PR, it just adjusts testnet `k` to be such that epoch is one day (this is used in modern testnet staging)

File: core/Pos/Core/Constants.hs [hunk 2]
```diff
@@ -75,6 +77,8 @@ data CoreConstants = CoreConstants
       -- | Size of mem pool will be limited by this value muliplied by block
       -- size limit.
     , ccMemPoolLimitRatio          :: !Word
+      -- | Suffix for genesis.bin files
+    , ccGenesisBinSuffix           :: ![Char]
```
Comment (neongreen) [inline core/Pos/Core/Constants.hs]: Why not `String`?

File: genesis-info/tn.log [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+                                 Dload  Upload   Total   Spent    Left  Speed
+
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 2655k    0 2655k    0     0  4022k      0 --:--:-- --:--:-- --:--:-- 4023k
100 9150k    0 9150k    0     0  5508k      0 --:--:--  0:00:01 --:--:-- 5508k
100 14.8M    0 14.8M    0     0  5712k      0 --:--:--  0:00:02 --:--:-- 5711k
100 20.8M    0 20.8M    0     0  5822k      0 --:--:--  0:00:03 --:--:-- 5822k
100 22.0M    0 22.0M    0     0  5827k      0 --:--:--  0:00:03 --:--:-- 5826k
+
+Running command: /nix/store/1yhsgimnbfiyqssgmcfcg67s8if8hvg0-cardano-sl-0.4.4/bin/cardano-keygen -f /home/ielectric/dev/serokell/daedalus/cardano-sl/scripts/generate/../../secrets/secret-{}.key -m 14 -n 12000 --richmen-share 0.94 --testnet-stake 19072918462000000 --utxo-file /home/ielectric/dev/serokell/daedalus/cardano-sl/scripts/generate/../avvm-files/utxo-dump-last-new.json --randcerts --blacklisted /home/ielectric/dev/serokell/daedalus/cardano-sl/scripts/generate/../avvm-files/full_blacklist.js --fake-avvm-seed-pattern avvm/fake-{}.seed --fake-avvm-entries 100
```
Comment (gromakovsky) [inline genesis-info/tn.log]: Why do we need anything before this line in this file in the repository?

File: core/constants.yaml [hunk 4]
```diff
@@ -121,31 +123,38 @@ qanet: &qanet
...
+testnet_wallet: &testnet_wallet
+  <<: *testnet
+  applicationName: csl-daedalus
 
+testnet_staging_wallet: &testnet_staging_wallet
+  <<: *testnet_staging
+  applicationName: csl-daedalus
+
+qanet_wallet: &qanet_wallet
+  <<: *qanet
```
Comment (gromakovsky) [inline core/constants.yaml]: I must confess that I don't understand anything in this file. Hope it's ok.

File: genesis-info/tn.log (hunk 3 above)
Comment (georgeee) [inline genesis-info/tn.log]: It's being logged automatically, we didn't want to write complex scripts for handling that (i.e. that will log accurately). This file is for double-checking mostly

File: core/Pos/Core/Constants.hs (hunk 2 above)
Comment (georgeee) [inline core/Pos/Core/Constants.hs]: B/c it's not present in `universum`?

Comment (gromakovsky) [inline core/Pos/Core/Constants.hs]: Are you from 2016? 😱
